test/decode-events.test.ts

```diff
@@ -0,0 +1,65 @@
+import { expect } from "chai";
+import { starknet } from "hardhat";
+import { StarknetContract, StarknetContractFactory } from "hardhat/types/runtime";
+import { TIMEOUT } from "./constants";
+import { BigNumber } from "ethers";
+import { expectFeeEstimationStructure } from "./util";
+
+/**
+ * Receives a hex address, converts it to bigint, converts it back to hex.
+ * This is done to strip leading zeros.
+ * @param address a hex string representation of an address
+ * @returns an adapted hex string representation of the address
+ */
+ function adaptAddress(address: string) {
+  return "0x" + BigInt(address).toString(16);
+}
+
+const OK_TX_STATUSES = ["PENDING", "ACCEPTED_ON_L2", "ACCEPTED_ON_L1"]
+
+/**
+ * Expects address equality after adapting them.
+ * @param actual 
+ * @param expected 
+ */
+function expectAddressEquality(actual: string, expected: string) {
+  expect(adaptAddress(actual)).to.equal(adaptAddress(expected));
+}
+
+describe("Starknet", function () {
+  this.timeout(TIMEOUT);
+
+  let contractFactory: StarknetContractFactory;
+  let eventsContractFactory: StarknetContractFactory;
+
+  before(async function() {
+    // assumes contract.cairo and events.cairo has been compiled
+    contractFactory = await starknet.getContractFactory("contract");
+    eventsContractFactory = await starknet.getContractFactory("events");
+  });
+
+  it("should decode events successfully", async function() {
+    const contract = await eventsContractFactory.deploy();
+    
+    const txHash = await contract.invoke("increase_balance", { amount: 10 });
+    const tx = await starknet.getTransaction(txHash);
```

Comment:
You can remove lines 45-49, this test should only test if the decoding is done properly - assertion of getTransaction response is already done in a separate test.